framework/base/Model.php

```diff
@@ -917,9 +929,15 @@ public static function validateMultiple($models, $attributeNames = null)
      */
     public function fields()
     {
-        $fields = $this->attributes();
+        static $fields = null;
+
+        if ($fields === null) {
+            $attributes = $this->attributes();
```

Comment:
No, if `attributes()` return varyon the model state, which can be (see #1318) method will produce invalid result.